### DISCUSSION TITLE
Easy way to convert a model into a collection?

### DISCUSSION THREAD

**Discussion by daemuth** [manyfold3d/manyfold#3127] [Support]
Hi!

I have about 2tb of models sorted in a specific folder structure:

`{creator}/{collection}/`

and within that `{collection}` folder I have folders for models, files, etc.

When I import my files into manyfold, the collections are getting recognised as a single model. 

Is there a way, even if it's a hack, to convert top-level models into collections? I **don't** want to merge everything into a single model.

Thanks!

**Comment by daemuth** [comment]
I think I found a bug in the meantime as well:

From the bulk edit screen: (`/models/N7TBt93f/model_files/bulk_edit`)

when I select a couple of files to extract into a new model, it gives me an error: 
![image](https://github.com/user-attachments/assets/2a5f4184-9dad-4382-a16c-7393363503c4)

So even if I'd manually go through my entire collection (which is not feasible on this scale), I couldn't do it.

**Comment by Floppy** [comment]
What's the folder structure under the collection path? On first scan, it should pick up any folder with indexable files in as a separate model...

**Comment by daemuth** [comment]
Folder struct of  `{creator}\2019_12 - The Sun Elves of Sol'Sylvaine`:
![image](https://github.com/user-attachments/assets/af94890e-bb0a-4e4c-aa24-71f5f333e10b)

Result in Manyfold:
![image](https://github.com/user-attachments/assets/642e1728-fac5-462b-a170-1ea5e4478117)

**Comment by daemuth** [comment]
Wait, nevermind, I think I might have subdirectories in subdirectories that I wasn't aware of. I'll check tomorrow and get back to you!

**Reply by Jeppedy** [reply]
I certainly have subs in subs, like Gridfinity/bins/standard
Gridfinity/bins/custom
Gridfinity/bases/thin
Gridfinity/bases/magnets
...

How is that supposed to work in MF?
Is there something I can do so it builds collections down to the last folder?
(Can I even have collections in collections?)

**Reply by Floppy** [reply]
You can indeed have nested collections, though it won't build them automatically I'm afraid. Easiest way is probably to manually create the collections, and put the models in with bulk edit.
